src/development/coding-securely-personal-data.md

```diff
@@ -0,0 +1,218 @@
+
+It is always easier not to log any personal data.
+
+Considerations:
+
+* Do not pass personal data directly through to anything that logs to a file (or any other destination).
+  * This includes System.out/System.err calls
+* Exclude or redact personal data in toString() methods.
+  * It is very easy to pass a composite object into a log statement and not to realise that nested, personal data will be logged.
+  * Use a library such as bsa-redact for a simple approach to redacting personal data
+* Make use of log appenders that recognise personal data prefixes in log statements and obscure/remove them
+  * This technique is not recommended as a primary mechanism as it relies on logging the data with appropriate markup to identify the personal data
+  * It can be useful to catch unexpected logging of private data such as usernames and passwords
+* Do not log Personal Data at debug level
+  * Debug level can be turned on in production environments in order to track down production issues
+  * It is always safer not to log any personal data
+  * If debug logging of personal data is required, use a different technique, such as turning off data redaction with a system property.
+* Review your logs
+  * Prior to going live with software, always take time to review the logs to check personal data is not being logged
+  * Review logs on every release to ensure Personal Data doesn't creep in somehow.
+
+#### Google Analytics (or equivalent)
+
+The [terms and conditions of Google Analytics (GA)][ga_terms_and_conditions] are very clear on the prohibition of sending Personally Identifiable Information to their servers (see section 7).
+
+Considerations:
+
+* Never use Personal Data in a URL
+* Sophisticated use of GA increases the risk of accidentally sending Personal Data
+  * If you are using Tag Manager or equivalent to send additional data to GA, ensure it does not include Personal Data
+
+#### Email
+
+Sending an automated email obviously includes the Personal Data of email address. However, there are risks associated with email that must be considered with respect to securing Personal Data
+
+Considerations
+
+* Email is insecure
+  * Do you know if the recipient's email provider uses secure transport and storage.?
+  * Do you know if the recipient will download the email to a secure location?
+  * Some email providers guarantee secure transport and storage when sent internally (e.g. NHS Mail). This doesn't guarantee the email is secure once it has been delivered
+* Keep Personal Data to a minimum in emails.
+* Always seek advice from Information Governance when sending automated emails
+
+### Is Personal Data secured?
+
+Transport and storage of Personal Data must be secure. It  is beyond the scope of this guidance to go into depth on this topic.
```

Comment:
Double space. `It  is beyond the scope `

Reply:
fixed, and search replaced all markdown for double spaces